workspaces/ui-v2/src/optic-components/hooks/util/useDebouncedFn.ts

```diff
@@ -0,0 +1,9 @@
+import { useMemo } from 'react';
+import debounce from 'lodash.debounce';
+
+export const useDebouncedFn = <T extends (...args: any) => any>(
```

Comment:
There's an existing useDebounce hook somewhere in the code - but I think it's used to debounce changes to a state object - in this case we want a fn to only run in a certain instance (and we need to wrap it in a useMemo otherwise the identity between renders change and it doesn't actually debounce